src/client/tokens.js

```diff
@@ -0,0 +1,299 @@
+const msal = require('@azure/msal-node')
+const XboxLiveAuth = require('@xboxreplay/xboxlive-auth')
+const fs = require('fs')
+const path = require('path')
+const fetch = require('node-fetch')
+const authConstants = require('./authConstants')
+
+function debug (...message) {
```

Comment:
Use const debug = require('debug')('minecraft-protocol') instead for items like sending out the data recieved by msal, xla and node-fetch. Anything that has to be sent should use console log like asking a user to complete a 2fa step.